wordpress-mobile/WordPress-Android
# Enable auto-uploading/remote-auto-saving for locally changed pages

Fixes #11126 

Enables automatic upload/remote-auto-save for pages with local changes. This feature was disabled before we released Offline Posting. This PR technically just reverts https://github.com/wordpress-mobile/WordPress-Android/commit/0a123667f9a2bccebc199feed9c1fcabc7afcc2a with minor modifications. It also introduces a new tracking event.

To test:
1. Go to Pages List
2. Click on an existing draft
3. Turn on airplane mode
4. Click on Publish
5. Turn off airplane mode
6. Notice the page gets auto-uploaded (check it's published on the blog/calypso)
7. Verify `auto_upload_page_invoked` was tracked with UPLOAD action
------------------------
1. Go to Pages List
2. Click on an existing draft
3. Turn on airplane mode
4. Make a change to the content
5. Press back (don't publish it)
5. Turn off airplane mode
6. Notice the page gets remote-auto-saved (check an unpublished revision is visible in Calypso)
7. Verify `auto_upload_page_invoked` was tracked with REMOTE_AUTO_SAVE action
-----------
We'll do thorough tests before we merge this whole feature into `develop`. So there is no need to perform any deep tests.

PR submission checklist:
